lib/ecto/datetime.ex

```diff
@@ -50,6 +50,8 @@ defmodule Ecto.DateTime.Utils do
   """
   def usec("." <> rest) do
     case parse(rest, "") do
+      {int, rest} when byte_size(int) > 6 and is_iso_8601(rest) ->
+        usec("." <> String.slice(int, 0, 6))
```

Comment:
as an alternative, we could just alter the clause below to match `> 1` instead of `in 1..6`, and then take a `String.slice(int, 0, 6)` or something prior to doing any padding or `to_integer` parsing.